### DISCUSSION TITLE
Ubuntu 22.04 firefox snap - won't install

### DISCUSSION THREAD

**Discussion by guygross18** [utmapp/UTM#5834] [Linux]
Hi, I've been trying to create an Ubuntu 22.04 vm on a 2021 MacBook, running M1 Pro. 
After successfully installing Ubuntu live server, and upgrading all the packages, I use "$sudo apt install ubuntu-desktop" to install the GUI. Every single time the installer hangs up while installing the Firefox snap. Is there a fix to this? what am I doing wrong?

**Comment by wrmack** [comment]
If you are using Apple virtualization (and not QEMU) then the problem might be the Linux kernel.  I had a lot of hangs with 22.04 on M1 Pro.  I recommend 23.10 - it is much more stable.

**Reply by wrmack** [reply]
Regardless - I recall the Firefox snap taking an inordinate amount of time to install. Didn't hang - just took a very long time.

**Reply by guygross18** [reply]
I'm using QEMU, not apple. 
It's stuck on 1% for several hours, so I don't think it's stuck.. just dead

**Reply by guygross18** [reply]
also, I can't quite the console using ctrl+c

**Reply by wrmack** [reply]
Not sure what to suggest.  Some possibilities:
- on your M1 pro check Activity Monitor: if CPU usage is maxed out this would verify it has hung, possibly due to a kernel panic
- I don't know any way to quit other than closing the window, which risks corrupting the file system - I am now used to running fsck at least once each session.
- once booted back into the server, have a look at the logs to see what might have happened: 
```bash
journalctl --system # will display all logs; Shft-G to go to end
journalctl --system -p err # will display errors only
```
- if I find there is an EXT4-fs error I reinstall the package indicated by the 'com' entry - the rationale being that Ubuntu might have trouble with that package if part of it is corrupted
- in order to run fsck (file system check) the VM needs to be unmounted - I have added an entry to the grub menu to boot into the live server iso in order to do this

Hope that helps.

**Reply by EshanDev27** [reply]
Did you find the solution, cause even im having the same problem

**Reply by wrmack** [reply]
I discovered there was a general issue with Apple virtualization and the installation of linux with regard to writing to disk. I imagine that is now well fixed in UTM but I changed to using tart to set up an Ubuntu VM. In case you are interested: https://gist.github.com/wrmack/c4e43aa7b6b14b9601cc2e4d6d575ac7 

I still run a Windows 11 VM using UTM.

However I think the issue with installing Firefox was probably a different issue.  A Google search implies a problem with servers might be the issue. In regard to Firefox in Ubuntu I have switched to Vivaldi as a browser - I needed something lightweight. Vivaldi seems to be able to do everything Firefox can and more.

**Comment by KingOfMac** [comment]
Run `sudo apt --fix-broken install` then retry
